src/mono/dbi/cordb-frame.cpp

```diff
@@ -0,0 +1,534 @@
+        m_dbgprot_buffer_free(&localbuf);
+
+        ReceivedReplyPacket* received_reply_packet = conn->GetReplyWithError(cmdId);
+        CHECK_ERROR_RETURN_FALSE(received_reply_packet);
+        MdbgProtBuffer* pReply = received_reply_packet->Buffer();
+
+        m_nFrames  = m_dbgprot_decode_int(pReply->p, &pReply->p, pReply->end);
+        m_ppFrames = (CordbNativeFrame**)malloc(sizeof(CordbNativeFrame*) * m_nFrames);
+
+        for (int i = 0; i < m_nFrames; i++)
+        {
+            int frameid   = m_dbgprot_decode_int(pReply->p, &pReply->p, pReply->end);
+            int methodId  = m_dbgprot_decode_id(pReply->p, &pReply->p, pReply->end);
+            int il_offset = m_dbgprot_decode_int(pReply->p, &pReply->p, pReply->end);
+            int flags     = m_dbgprot_decode_byte(pReply->p, &pReply->p, pReply->end);
+
+            CordbNativeFrame* frame = new CordbNativeFrame(conn, frameid, methodId, il_offset, flags, m_pThread);
+            frame->InternalAddRef();
+            m_ppFrames[i] = frame;
+        }
+
+        if (!m_pThread->GetStepper())
+            m_pThread->SetRegisterSet(new CordbRegisterSet(conn, 0, 0));
+
+        *pcelt = m_nFrames;
+    }
+    EX_CATCH_HRESULT(hr);
+    return hr;
+}
+
+HRESULT STDMETHODCALLTYPE CordbFrameEnum::QueryInterface(REFIID riid, void** ppvObject)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbFrameEnum - QueryInterface - NOT IMPLEMENTED\n"));
+    return E_NOTIMPL;
+}
+
+CordbJITILFrame::CordbJITILFrame(
+    Connection* conn, int frameid, int methodId, int il_offset, int flags, CordbThread* thread)
+    : CordbBaseMono(conn)
+{
+    this->m_debuggerFrameId  = frameid;
+    this->m_debuggerMethodId = methodId;
+    this->m_ilOffset         = il_offset;
+    this->m_flags            = flags;
+    this->m_pThread          = thread;
+}
+
+HRESULT STDMETHODCALLTYPE CordbJITILFrame::GetChain(ICorDebugChain** ppChain)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbFrame - GetChain - NOT IMPLEMENTED\n"));
+    return E_NOTIMPL;
+}
+
+HRESULT STDMETHODCALLTYPE CordbJITILFrame::GetCode(ICorDebugCode** ppCode)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbFrame - GetCode - NOT IMPLEMENTED\n"));
+    return E_NOTIMPL;
+}
+
+HRESULT STDMETHODCALLTYPE CordbJITILFrame::GetFunction(ICorDebugFunction** ppFunction)
+{
+    CordbFunction* func = conn->GetProcess()->FindFunction(m_debuggerMethodId);
+    if (!func)
+    {
+        func = new CordbFunction(conn, 0, m_debuggerMethodId, NULL);
+    }
+    func->QueryInterface(IID_ICorDebugFunction, (void**)ppFunction);
+    LOG((LF_CORDB, LL_INFO1000000, "CordbFrame - GetFunction - IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT STDMETHODCALLTYPE CordbJITILFrame::GetFunctionToken(mdMethodDef* pToken)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbFrame - GetFunctionToken - NOT IMPLEMENTED\n"));
+    return E_NOTIMPL;
+}
+
+HRESULT STDMETHODCALLTYPE CordbJITILFrame::GetStackRange(CORDB_ADDRESS* pStart, CORDB_ADDRESS* pEnd)
+{
+    *pStart = 4096;
```

Comment:
I would use a constant to describe what are these values

Reply:
Changed and added a comment.